.travis.yml

```diff
@@ -1,12 +1,50 @@
 language: python
-sudo: enabled
+sudo: required
+dist: trusty
+addons:
+  ssh_known_hosts: localhost
+  postgresql: "9.5"
+cache:
+  pip: true
+  directories:
+    - assembl/static/js/node_modules
+    - assembl/static/widget/card/bower_components
+    - assembl/static/widget/creativity/bower_components
+    - assembl/static/widget/session/bower_components
+    - assembl/static/widget/share/bower_components
+    - assembl/static/widget/video/bower_components
+    - assembl/static/widget/vote/bower_components
+    - assembl/static2/node_modules
 before_install:
- - sudo apt-get -qq update
+ # The following line is needed, otherwise the set_file_permissions fab command yells error "Operation not permitted"
+ - sudo usermod -a -G www-data $USER
  - sudo apt-get install -y fabric git openssh-server
- - fab -c configs/develop.rc install_single_server
- - fab -c configs/develop.rc bootstrap_from_checkout
-install: true
+ - ssh-keygen -P '' -f ~/.ssh/id_rsa && cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+ - echo "_extends = ./testing.rc" > configs/travis.rc; echo "_gitbranch = $TRAVIS_COMMIT" >> configs/travis.rc;
+ # The following line produces warning "uses weak digest algorithm" during its `apt-get update`
+ - fab -c configs/travis.rc install_assembl_server_deps
+ - fab -c configs/travis.rc install_redis
+ - fab -c configs/travis.rc install_memcached
+install:
+ # The set_file_permissions fabric command needs to be encapsulated into this sudo su command, because in before_install we have added $USER in www-data group but have not logged out and logged in again for changes to take effect in this terminal
+ - sudo -E su $USER -c 'fab -c configs/travis.rc set_file_permissions'
+ # TODO: fix fabfile's check_and_create_database_user command which produces an error when run by travis. For now we instead run the following command
+ - pip install psycopg2
```

Comment:
That and a few other things could be part of install_basetools
I guess you decided you did not need install_single_server ? So no testing elasticsearch?

Reply:
There is no elasticsearch tests currently.